legal-api/src/legal_api/services/authz.py

```diff
@@ -877,8 +877,24 @@ def has_notice_of_withdrawal_filing_blocker(business: Business):
     if business.admin_freeze:
         return True
 
-    filing_statuses = [Filing.Status.DRAFT.value,
-                       Filing.Status.PENDING.value,
+    # Get request method and filing id from request context
```

Comment:
These checks are not required

Reply:
Notice of Withdrawal filing is supposed to withdrawal future effective filings. That's why there is a special check for notice of withdrawal. (this was the previous ticket for allowable actions for NoW https://github.com/bcgov/entity/issues/23348, and this was the PR https://github.com/bcgov/lear/pull/3014)

The issue this PR is trying to address is that with the previous implementation, we cannot update the NoW draft using  `PUT` method with endpoint `/api/v2/businesses/:identifier/filings/:filing_id?draft=true`. And this PR is trying to make this happen and keep all other checks remain the same.

Reply:
I did not mean that this function is not required. We only need this check
```
    def has_notice_of_withdrawal_filing_blocker(business: Business):
        """Check if there are any blockers specific to Notice of Withdrawal."""
        if business.admin_freeze:
            return True
      
        now = datetime.now(timezone.utc)
        paid_filings = Filing.get_filings_by_status(business.id, [Filing.Status.PAID.value])
        return not any(f.effective_date and f.effective_date > now for f in paid_filings)
```

Reply:
Thanks for clarification, I tried that, but it ended up to allow filing multiple drafts
![202412191612](https://github.com/user-attachments/assets/7b5dbb6a-12e0-4beb-996a-147e5bf46b5a)

Reply:
Ok I missed that scenario. Let me check

Reply:
Use `is_ignore_draft_blockers` (https://github.com/bcgov/lear/blob/main/legal-api/src/legal_api/services/authz.py#L765) similar to `has_blocker_filing` function

Reply:
Good call, gonna try it out.

Reply:
It works, thanks for the inspiration, love it